### PR TITLE
Make extract_volume work with closed and mixed intervals and allow nearest value selection

### LINKED ISSUE #1926: Behaviour of `extract_volume` - allow extraction on closed or mixed intervals
**Is your feature request related to a problem? Please describe.**
Currently, `extract_volume` is just able to extract a range in the z-axis with an open interval given by `(z_min, z_max)`. It would be useful to be able to also extract closed or mixed intervals `[z_min, z_max]`, `[z_min, z_max)`, `(z_min, z_max]` .

It would also be useful to have the option to extract a range considering the closest coordinate values to z_min and z_max. For example, in a coordinate with points:

```
lev = 0.505760014057159, 1.55585527420044, 2.66768169403076, 
    3.8562798500061, 5.14036130905151, 6.54303359985352, 8.09251880645752, 
    9.82275009155273, 11.7736797332764, 13.9910383224487, 16.5253219604492, 
    19.429801940918, 22.7576160430908, 26.5583000183105, 30.8745613098145, 
    35.7402038574219, 41.1800231933594, 47.2118949890137, 53.8506355285645, 
    61.1128387451172, 69.0216827392578, 77.6111602783203, 86.9294281005859, 
    97.0413131713867, 108.030281066895, 120, 133.075820922852, 147.40625, 
    163.164459228516, 180.549926757812, 199.789962768555, 221.141174316406, 
    244.890625, 271.356384277344, 300.887512207031, ...
```
if `z_max: 300`, it should be able to select the last level as well because 300.89 is closer to 300 than the previous level of 271.36 .

These changes could be implemented so that the current behaviour is the default one, in order to make them backward compatible.

**Would you be able to help out?**
Yes

### PR DESCRIPTION
**Pull request by sloosvel**
## Description

This PR expands the capabilities of `extract_volume` so that it works with closed and mixed intervals. It also adds the possibility to perform the extraction taking into account the coordinate values.

There is also a small fix in the documentation regarding a link that was not pointing to the right function.

Closes #1926 

Link to documentation: https://esmvaltool--1930.org.readthedocs.build/projects/ESMValCore/en/1930/recipe/preprocessor.html#extract-volume

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
~- [ ] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly~
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Review by valeriupredoi — APPROVED** [review]
nice! I'd like to see the explanation of whan a left/right open interval means in terms of the operation itself ie closed: `<=/>=` and open:`</>` but that's just nitpicking at this late stage before the freeze :beer:

**Comment by sloosvel** [comment]
No worries I can add now a short explanation

**Comment by valeriupredoi** [comment]
if you could that'd be nice - pls merge main into here too, hoping no merge conflicts. Cheers Saskia :beer:

File: doc/recipe/preprocessor.rst [hunk 1]
```diff
@@ -1879,9 +1879,14 @@ The ``_volume.py`` module contains the following preprocessor functions:
...
 Extract a specific range in the `z`-direction from a cube. The range is given as an interval
-that can be open ``(z_min, z_max)``, closed ``[z_min, z_max]``, left closed ``[z_min, z_max)``
-or right closed ``(z_min, z_max]``. The extraction is performed by applying a constraint on the
-coordinate values, without any kind of interpolation. 
+that can be:
+
+* open ``(z_min, z_max)``, in which the extracted range does not includes ``z_min`` nor ``z_max``.
+* closed ``[z_min, z_max]``, in which the extracted includes both ``z_min`` and ``z_max``.
+* left closed ``[z_min, z_max)``, in which the extracted range includes ``z_min`` but not ``z_max``.
+* right closed ``(z_min, z_max]``, in which the extracted range includes ``z_max`` but not ``z_min``.
```
Comment (valeriupredoi) [inline doc/recipe/preprocessor.rst:1887]: wonderful, cheers, Saskia!

**Pull request merged**